openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+    )
+    tolerationSeconds: Optional[int] = Field(
+        None,
+        description='TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict). Zero and negative values will be treated as 0 (evict immediately) by the system.',
+    )
+    value: Optional[str] = Field(
+        None,
+        description='Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty, otherwise just a regular string.',
+    )
+
+
+class DataProtectionApplicationConfigurationVeleroPodConfig(BaseModel):
+    labels: Optional[Dict[str, str]] = Field(None, description='labels to add to pods')
+    nodeSelector: Optional[Dict[str, str]] = Field(
+        None,
+        description='nodeSelector defines the nodeSelector to be supplied to Restic podSpec',
+    )
+    resourceAllocations: Optional[Any] = Field(
+        None,
+        description='resourceAllocations defines the CPU and Memory resource allocations for the restic Pod',
+    )
+    tolerations: Optional[List[TolerationModel]] = Field(
+        None,
+        description='tolerations defines the list of tolerations to be applied to Restic daemonset',
+    )
+
+
+class DataProtectionApplicationConfigurationVelero(BaseModel):
+    customPlugins: Optional[List[CustomPlugin]] = Field(
+        None,
+        description='customPlugins defines the custom plugin to be installed with Velero',
+    )
+    defaultPlugins: Optional[List[str]] = None
+    featureFlags: Optional[List[str]] = Field(
+        None,
+        description='featureFlags defines the list of features to enable for Velero instance',
+    )
+    logLevel: Optional[LogLevel] = Field(
+        None,
+        description='Velero server’s log level (use debug for the most logging, leave unset for velero default)',
+    )
+    noDefaultBackupLocation: Optional[bool] = Field(
+        None,
+        description='If you need to install Velero without a default backup storage location noDefaultBackupLocation flag is required for confirmation',
+    )
+    podConfig: Optional[DataProtectionApplicationConfigurationVeleroPodConfig] = Field(
+        None, description='Pod specific configuration'
+    )
+    restoreResourcesVersionPriority: Optional[str] = Field(
+        None,
+        description='restoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with EnableAPIGroupVersions feature flag Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag',
+    )
+
+
+class DataProtectionApplicationConfiguration(BaseModel):
```

Comment:
please rename DataProtectionApplicationConfiguration -> ApplicationConfiguration

Reply:
don't see you renamed it, so I unresolved the thread